app/domain/operations/financial_assistance/on_determination/families/create_or_update.rb

```diff
@@ -0,0 +1,311 @@
+          def validate_application(application)
+            if application.is_a?(::FinancialAssistance::Application)
+              if application.family.is_a?(::Family)
+                Success([application, application.family])
+              else
+                Failure("Invalid Family for given application with hbx_id: #{application.hbx_id}")
+              end
+            else
+              Failure('Invalid application type. Expected FinancialAssistance::Application.')
+            end
+          end
+
+          # Creates or updates people for all applicants in the application
+          #
+          # @param application [FinancialAssistance::Application] the financial assistance application
+          # @return [Dry::Monads::Result] Success with hash of applicant_id => person or Failure with results hash
+          def create_or_update_people(application)
+            final_result_success = true
+            results = application.applicants.inject({}) do |result_hash, applicant|
+              person_result = create_or_update_person_by(applicant)
+              final_result_success = false if person_result.failure?
+              result_hash[applicant.id] = person_result.success
+              result_hash
+            end
+
+            final_result_success ? Success(results) : Failure(results)
+          end
+
+          # Creates or updates a person based on an applicant
+          #
+          # @param applicant [FinancialAssistance::Applicant] the applicant to create or update a person for
+          # @return [Dry::Monads::Result] Success with person or Failure with error message
+          def create_or_update_person_by(applicant)
+            ::Operations::FinancialAssistance::OnDetermination::People::CreateOrUpdate.new.call(applicant: applicant)
+          end
+
+          # Creates or updates relationships for the primary applicant
+          #
+          # @param people_result [Hash] hash of applicant_id => person
+          # @param application [FinancialAssistance::Application] the financial assistance application
+          # @return [Dry::Monads::Result] Success with message or Failure with error message
+          def create_or_update_primary_relationships(people_result, application)
+            primary_applicant = application.primary_applicant
+            primary_person = people_result[primary_applicant.id]
+
+            application.relationships.where(applicant_id: primary_applicant.id).each do |rel|
+              existing_rel = primary_person.person_relationships.where(relative_id: people_result[rel.relative_id].id).first
+
+              if existing_rel.present?
+                existing_rel.kind = rel.kind
+              else
+                primary_person.person_relationships.build(kind: rel.kind, relative_id: people_result[rel.relative_id].id)
+              end
+            end
+
+            primary_person.save!
```

Comment:
adding a failure case if this fails to save might be overkill but might be helpful